test/unittest/rtps/domain/RTPSDomainTests.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <fastdds/LibrarySettings.hpp>
+#include <fastdds/rtps/RTPSDomain.h>
+
+/**
+ * This test checks the getter and setter for the library settings in the RTPS layer.
+ */
+TEST(RTPSDomainTests, library_settings_test)
+{
+    eprosima::fastdds::LibrarySettings library_settings;
+    EXPECT_TRUE(eprosima::fastrtps::rtps::RTPSDomain::get_library_settings(library_settings));
+    // Get LibrarySettings default values
+#if HAVE_STRICT_REALTIME
+    EXPECT_EQ(eprosima::fastdds::INTRAPROCESS_OFF, library_settings.intraprocess_delivery);
+#else
+    EXPECT_EQ(eprosima::fastdds::INTRAPROCESS_FULL, library_settings.intraprocess_delivery);
+#endif
+    library_settings.intraprocess_delivery = eprosima::fastdds::INTRAPROCESS_USER_DATA_ONLY;
+    // Setting the library settings within an empty RTPSDomain shall return true
+    EXPECT_TRUE(eprosima::fastrtps::rtps::RTPSDomain::set_library_settings(library_settings));
+    EXPECT_TRUE(eprosima::fastrtps::rtps::RTPSDomain::get_library_settings(library_settings));
+    EXPECT_EQ(eprosima::fastdds::INTRAPROCESS_USER_DATA_ONLY, library_settings.intraprocess_delivery);
+    // Create RTPSParticipant
+    eprosima::fastrtps::rtps::RTPSParticipantAttributes part_attr;
+    eprosima::fastrtps::rtps::RTPSParticipant* participant =
+            eprosima::fastrtps::rtps::RTPSDomain::createParticipant(0, part_attr);
+    ASSERT_NE(nullptr, participant);
```

Comment:
I ran the test and passed, everything OK. I just got this warning
 ```
Uninteresting mock function call - returning directly. Function call: init_thread() NOTE: You can safely ignore the above warning unless this call should not happen.  
```
I think that this warning can be just ignored, but if this is not the case using EXPECT_CALL should salve the problem

Reply:
It is not possible to use the EXPECT_CALL because in order to use that macro, access to the mocked class should be possible. In this case, the API being called is `RTPSDomain::createParticipant`. Internally, it is this API that created the RTPSParticipantImpl and the corresponding ResourceEvent instance that would need to be used. So we would have to keep the warning for the moment.